src/MongoDB.Driver.Legacy/Builders/IndexKeysBuilder.cs

```diff
@@ -261,6 +271,30 @@ public override BsonDocument ToBsonDocument()
             return _document;
         }
 
+        /// <summary>
+        /// Sets a wildcard key to the index. If the wildcard name is empty, the generated key will be for the whole collection.
+        /// </summary>
+        /// <param name="name">The wildcard key.</param>
+        /// <returns>The builder (so method calls can be chained).</returns>
+        public IndexKeysBuilder Wildcard(string name = null)
```

Comment:
Why did you object to the default value being "$**"?

Reply:
Good point. Done.

Reply:
an interesting fact (at least for me), If I set name to "$**" by default instead just `null`, I cannot set any parameter when I call this method outside even if this parameter is also default method value.
So, it doesn't work:

    public IndexKeysBuilder Wildcard(string name = "$**")
    {
           // here we will have null instead `$**`
    }
    public void SomeOtherMethod(string name = null)
    {
           Wildcard(name);
    }

Fixed.

Reply:
I don't understand the problem. Can you elaborate?

Reply:
Why isn't the default value for SomeOtherMethod also "$**"?

I think the problem is not with using "$**" as the default value but with some other mismatch.

I still think the most appropriate default value is "$**" but am willing to be flexible.

Reply:
I think you are confusing:

    builder.Wildcard(); // compiler replaces missing argument with default value

with

    builder.Wildcard(null); // no replacement because an explicit value was provided

Reply:
to have the name parameter with default value `"$**"`, we should not pass any parameter to this method. So, to make it possible, the above code from my comment should be rewritten to this one:

    public IndexKeysBuilder Wildcard(string name = "$**")
    {
         // here we will have `$**`
    }
    public void SomeOtherMethod(string name = null)
    {
         if (name == null) 
         {
             Wildcard();
         }
         else
         {
             Wildcard(name);
         }
    }

Reply:
I'm not sure that having additional `if` in any place when we call low level `Wildcard` method  worth it

Reply:
Go ahead and use null as the default value if you wish